File: backend/Sonic.Application/Users/DTOs/GetCurrentUserResponse.cs
namespace Sonic.Application.Users.DTOs;

public sealed class GetCurrentUserResponse
{
    public string Id { get; init; } = default!;
    public string Email { get; init; } = default!;
    public string DisplayName { get; init; } = default!;
    public string? Bio { get; init; }
    public string? JobRole { get; init; }
    public List<string> Interests { get; init; } = new();
    public string? AvatarUrl { get; init; }
    public string Role { get; init; } = default!;
    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }
}

File: frontend/sonic-web/src/app/shared/contracts/user/get-current-user-response.ts
export interface GetCurrentUserResponse {
    id: string;
    email: string;
    displayName: string;
    bio?: string | null;
    jobRole?: string | null;
    interests: string[];
    avatarUrl?: string | null;
    role: string;
    createdAt: string;
    updatedAt: string;
}